web/static/js/reducers/panelSurvey.js

```diff
@@ -1,13 +1,25 @@
 // @flow
 import * as actions from '../actions/panelSurvey'
+import * as surveyActions from '../actions/surveys'
 import fetchReducer from './fetch'
 
 export const dataReducer = (state: PanelSurvey, action: any): PanelSurvey => {
   switch (action.type) {
+    case surveyActions.DELETED: return deleteSurvey(state, action)
     default: return state
   }
 }
 
+const deleteSurvey = (state, action) => {
```

Comment:
Can't we go fetch the panel Survey to the api like you did when creating a `newOcurrence`?

If the backend for some reason (like if is the only occurrence of a PS) does not deletes the occurence, the UI will not show it (when it should)

Reply:
The `survey.DELETED` action shouldn't be triggered if the survey isn't deleted. If it does, we have a problem in another place, and I don't get why to do defensive programming here. If the action is triggering badly, we'll have problems [here](https://github.com/instedd/surveda/blob/511c8e79df9cf40a9420539755ea48ac3f5b19ca/web/static/js/reducers/surveys.js#L7) also. It's almost exactly the same situation.

As I see it, what we do by fetching the panel survey after creating a new occurrence is a workaround. There we should (ideally) follow a redux flow similar to this and not the other way round.

Reply:
ok, if the action shouldn't be triggered if not _actually_ deleted then seems ok.